drivers/bluetooth/hci/ipm_stm32wb.c

```diff
@@ -327,17 +327,37 @@ static void start_ble_rf(void)
 		LL_RCC_ReleaseBackupDomainReset();
 	}
 
-	/* Select LSE clock */
-	LL_RCC_LSE_Enable();
-	while (!LL_RCC_LSE_IsReady()) {
-	}
+	do {
+#ifdef CONFIG_CLOCK_STM32_LSE
+		const u32_t alarm = k_uptime_get_32() + K_MSEC(RCC_LSE_TIMEOUT_VALUE);
+		/* Select LSE clock */
+		LL_RCC_LSE_Enable();
+		while (!LL_RCC_LSE_IsReady()) {
+			if (alarm < k_uptime_get_32())
+				goto init_LSI;
```

Comment:
I'm fine with adding LSI support (while I have questions around clock is precise enough for RF, I'll checking internally).
Though, I don't think enabling as LSE back up make much sense.
If LSE is available on HW and config LSE is enabled, then it should work. If not working, there is a HW issue to be checked. Can you explain  this choice?

Reply:
I'm fine to with busy loop on LSE failure if LSE is configured. 
Keeping device functional even if there is issues with the HW, seems to me a valuable choice. As Far as i know (from my ST contact person) only HSE is required for normal RF operations

Reply:
> Keeping device functional even if there is issues with the HW, seems to me a valuable choice.

For a product, it could be useful indeed. Though before this you want to know that you have an issue with your clock. It could be an optonnal feature, but then there is no reason to do it here and not on other places where clock is needed (everywhere).
Please use busy loop on failure.

Reply:
Make sense. I'd prefer to have some LOG_ERR for LSE failure and continue to work on LSI. But there is a big change that this error will be just omitted. Done